test/packaging/tar/bz2_packaging.py

```diff
@@ -43,20 +44,75 @@
 if not tar:
     test.skip_test('tar not found, skipping test\n')
 
-bz2 = test.where_is('bzip2')
-
 if sys.platform == 'win32':
-    # windows 10 causes fresh problems by supplying a tar, not bzip2
```

Comment:
Are you ignoring the git tar? Which could be used in the case where the windows tar doesn't support bz2?

Reply:
> Which could be used in the case where the windows tar doesn't support bz2?

The test, as written prior to changes, only checks the first tar executable found on the path.

Reply:
Locally annotated change (not pushed yet)
```
        windir = os.environ.get("SystemRoot")
        if not windir:
            windir = os.environ.get("windir")
        if not windir:  # <== ADDED
            windir = os.path.join(os.environ.get("SystemDrive", "C:") + os.path.sep, "Windows")  # <== ADDED
        expected = os.path.normcase(os.path.abspath(os.path.join(windir, "System32", "tar.exe")))
        if tar != expected:
            return False
```

Additional clause added to always test the target executable path against what should be the windows system32 path.

Reply:
Isn't windir always defined on windows?

Reply:
It *should* be.

Depends on your level of paranoia:
```
(3.12.9) S:\SCons\Test-WinSkip>type test-env.bat
@setlocal
@python -c "import os; print([os.environ.get(var) for var in ['SystemDrive', 'SystemRoot', 'windir']])"
@set SystemDrive=
@set SystemRoot=
@set windir=
@python -c "import os; print([os.environ.get(var) for var in ['SystemDrive', 'SystemRoot', 'windir']])"
@endlocal

(3.12.9) S:\SCons\Test-WinSkip>test-env
['C:', 'C:\\Windows', 'C:\\Windows']
[None, None, None]
```